unit-tests/shared/src/test/scala/javalib/net/InetAddressTest.scala

```diff
@@ -40,6 +40,10 @@ class InetAddressTest {
     assertFalse(all == null)
     assertTrue(all.length >= 1)
 
+    // TODO remove filter on main
+    for (alias <- all; if alias.isInstanceOf[Inet4Address])
+      assertTrue(alias.getCanonicalHostName().startsWith("localhost"))
```

Comment:
This test is broken for IPv6 in `0.4.x`, but I think it is fixed on `main` thanks to @LeeTibbert's work in https://github.com/scala-native/scala-native/issues/2623.

Reply:
Thank you for noticing the 0.4.n defect, and the 0.5.0 fix.  Makes me feel my time fixing
was useful, if not timely.

I believe the current (0.4.x, 0.5.0) status is that all of IPv6 and all of Udp is busted in 
javalib.  Both should be working in posixlib.  I do not know off the top of my head
if `socket` is available in clib.

I am currently working my way through exercising  & probing sockets etc, in at least 
posixlib.

Quickly, so as not to distract you, what are your use-concerns?   I can factor
those into my work plans/hopes.  Thanks.

Reply:
@LeeTibbert thanks for asking!

> Quickly, so as not to distract you, what are your use-concerns?

I'm currently focused on cross-building two I/O+networking libraries:
- https://github.com/comcast/ip4s
- https://github.com/typelevel/fs2

I will use them to cross-build projects such as:
- https://github.com/http4s/http4s
- https://github.com/tpolecat/skunk

Every issue/PR I've opened is a bug I discovered via an existing test suite in those projects. I am trying very hard to fix every problem at the source, instead of hacking around bugs which gives Scala Native a bad name and forces projects to do ugly contortions.

Until the 0.5.x series is release, ideally all these fixes can be targeted at 0.4.x, since the projects I mentioned will be stuck without Scala Native support until their test suites are passing.

Reply:
Thank you for the info and all your work.  Sounds like you will be giving SN network I/O
good exercise.  

```
 I am trying very hard to fix every problem at the source, instead of hacking around bugs which gives Scala Native a bad name and forces projects to do ugly contortions.
```
I'll sing in that choir!   Bogus stuff definitely puts potential devos/users off,
and quickly too.

Yell if there is something I can do to remove SN network blemishes.
Not much that can be done about posixlib IPv6 for SN 0.5.0,
to the best of my understanding of the situation.  Javalib IPv6
and/or UDP are do-able for 0.4.n but require a lot of work.
To my thinking, other project's failing unit-tests might justify that work, after
the already implemented parts of SN are passing.

I know that you are, correctly, focused on correctness but you might
be interested.  I have a PR in the chute which should simplify
some socket code paths even more.  I hope they eke out a few
cpu cycles of efficiency. Of course, "over-the-wire" costs will
almost always continue to predominate.

Shall I send a coffee subscription to fuel your drive?